src/context/locale/localeContextController/LocaleContextController.tsx

```diff
@@ -0,0 +1,16 @@
+import { useState } from 'react';
+import { IntlProvider } from 'react-intl';
+import { translations } from '../../../i18n/messages';
+import { defaultLocale } from '../defaultLocale';
+import { localeContext } from '../localeContext/localeContetx';
+import { LocaleContextControllerProps } from './LocaleContextController.types';
+
+export const LocaleContextController = ({ children }: LocaleContextControllerProps) => {
+  const [locale, setLocale] = useState(defaultLocale);
+
+  return (
+    <IntlProvider defaultLocale={defaultLocale} key={locale} locale={locale} messages={translations[locale]}>
+      <localeContext.Provider value={{ locale, setLocale }}>{children}</localeContext.Provider>
```

Comment:
Czy ten context jest w ogóle potrzebny? Obie rzeczy jesteś w stanie wyciągnąć z `IntlProvider`a przez `useIntl`

Reply:
Potrzebuję stanu, żeby zmieniać język, stąd potrzeba tego contextu na poziomie IntlProvidera. Szukam jak to ominąć ale nie umiem znaleźć nic sensownego. Jakaś wskazówka?

Reply:
A funkcji zmiany locale nie widzę w useIntl

Reply:
Ano fakt, źle spojrzałem. Jedno co, to można by jeszcze zaimplementować hooka `useLocale`, który będzie miał wszystko co jest w `useIntl` plus `setLocale` i jego wszędzie używać

Reply:
ok